src/contracts/core/AllocationManager.sol

```diff
@@ -430,25 +313,53 @@ contract AllocationManager is
         IStrategy strategy,
         uint16 numToComplete,
         uint192 nextIndex
-    ) internal view returns (uint64 freeMagnitudeToAdd, uint192 completed) {
-        uint256 pendingFreeMagnitudeLength = _pendingFreeMagnitude[operator][strategy].length;
+    ) internal returns (uint64 freeMagnitudeToAdd, uint192 completed) {
+        uint256 numDeallocations = _pendingDeallocationOperatorSets[operator][strategy].length;
         freeMagnitudeToAdd = 0;
-        while (nextIndex < pendingFreeMagnitudeLength && completed < numToComplete) {
-            PendingFreeMagnitude memory pendingFreeMagnitude = _pendingFreeMagnitude[operator][strategy][nextIndex];
-            // pendingFreeMagnitude is ordered by completableTimestamp. If we reach one that is not completable yet, then break
+        while (nextIndex < numDeallocations && completed < numToComplete) {
+            bytes32 opsetKey = _pendingDeallocationOperatorSets[operator][strategy][nextIndex];
+            MagnitudeInfo memory opsetMagnitudeInfo = _operatorMagnitudeInfo[operator][strategy][opsetKey];
+            // _pendingDeallocationOperatorSets is ordered by `effectTimestamp`. If we reach one that is not completable yet, then break
             // loop until completableTimestamp is < block.timestamp
-            if (block.timestamp < pendingFreeMagnitude.completableTimestamp) {
+            if (block.timestamp < opsetMagnitudeInfo.effectTimestamp) {
                 break;
             }
 
             // pending free magnitude can be added to freeMagnitude
-            freeMagnitudeToAdd += pendingFreeMagnitude.magnitudeDiff;
+            freeMagnitudeToAdd += uint64(uint128(-opsetMagnitudeInfo.pendingMagnitudeDiff));
+            _operatorMagnitudeInfo[operator][strategy][opsetKey] = MagnitudeInfo({
+                pendingMagnitudeDiff: 0,
+                currentMagnitude: opsetMagnitudeInfo.currentMagnitude - uint64(uint128(-opsetMagnitudeInfo.pendingMagnitudeDiff)),
+                effectTimestamp: 0
+            });
+
             ++nextIndex;
             ++completed;
         }
         return (freeMagnitudeToAdd, completed);
     }
 
+    /// @dev Fetch the operator's current magnitude, applying a pending diff if the effect timestamp is passed
+    /// @notice This may return something that is not recorded in state. Remember to store this updated value if needed!
+    function _getCurrentMagnitudeWithPending(address operator, IStrategy strategy, bytes32 operatorSetKey) internal view returns (MagnitudeInfo memory) {
+        MagnitudeInfo memory mInfo = _operatorMagnitudeInfo[operator][strategy][operatorSetKey];
```

Comment:
Not a fan of the naming, feel like we should be more explicit.

Reply:
What would you change it to?